app/frontend/components/pages/accounts/accounts.tsx

```diff
@@ -0,0 +1,286 @@
+    </div>
+  )
+}
+
+const Accounts = ({
+  accounts,
+  setAccount,
+  reloadWalletInfo,
+  showSendTransactionModal,
+  showDelegationModal,
+  shouldShowSendTransactionModal,
+  shouldShowDelegationModal,
+  selectedAccount,
+  totalWalletBalance,
+  totalRewardsBalance,
+}) => {
+  const accountInfos = Object.values(accounts)
+  const firstAddressPerAccount = accountInfos.map((e: any) => e.visibleAddresses[0].address)
+  const InfoAlert = () => (
+    <Fragment>
+      <div className="dashboard-column account sidebar-item info">
+        <Alert alertType="info sidebar">
+          <p>
+            <strong>Accounts</strong> offer a way to split your funds. You are able to delegate to
+            different pool from each account. Each account has different set of addresses and keys.
+          </p>
+        </Alert>
+      </div>
+      <div className="dashboard-column account sidebar-item info">
+        <Alert alertType="info sidebar">
+          <p>
+            Click explore/activate button to load data for related account. If you are using
+            hardware wallet, you will be requested to export public key.
+          </p>
+        </Alert>
+      </div>
+      <div className="dashboard-column account info">
+        <Alert alertType="warning sidebar">
+          <p>
+            This feature might not be supported on other wallets yet. If you decide to move your
+            funds to <strong>account</strong> other then <strong>account 0</strong>, you might not
+            see your funds in other wallets.
+          </p>
+        </Alert>
+      </div>
+    </Fragment>
+  )
+
+  return (
+    <Fragment>
+      {shouldShowSendTransactionModal && <SendTransactionModal />}
+      {shouldShowDelegationModal && <DelegationModal />}
+      <div className="dashboard-column account">
+        <div className="card account-aggregated">
+          <div className="balance">
+            <div className="item">
+              <h2 className="card-title small-margin">Total balance</h2>
+              <div className="balance-amount">
+                {printAda(totalWalletBalance as Lovelace)}
+                <AdaIcon />
+              </div>
+            </div>
+            <div className="item">
+              <h2 className="card-title small-margin">Total rewards balance</h2>
+              <div className="balance-amount">
+                {printAda(totalRewardsBalance as Lovelace)}
+                <AdaIcon />
+              </div>
+            </div>
+          </div>
+          <div className="refresh-wrapper">
+            <button className="button secondary balance refresh" onClick={reloadWalletInfo}>
+              Refresh
+            </button>
+          </div>
+        </div>
+        <div className="mobile">
+          <InfoAlert />
+        </div>
+        <div className="accounts-wrapper">
+          <div className="dashboard-column account list">
+            <div>
+              {range(0, Object.keys(accounts).length + 1).map(
```

Comment:
why the +1? Is there the assumption that the last element of accounts is always unused?

Reply:
We want to display (numberOfUsedAccounts + 1) accountTiles. Refactored.